### ISSUE TITLE
Coorelations and Principal Components Analysis

### ISSUE THREAD

**Issue by RobertBiegler** [jasp-stats/jasp-desktop#1987]
A student and I ran Bayesian correlations on the same data set, and got different results for the strongest correlations.  We tracked it to a difference between versions.  Versions 0.8.0.0 and 0.8.0.1 reported a Bayes factor of 4.760 +e37, while version 0.8.0.2 reports a Bayes factor of 2.198 e+306 on Pearson's correlations between A and D in the attached file "JaspTest.csv".  A disagreement by 269 orders of magnitude seems a tad excessive.  Which version can we believe, if any?

Also, version 0.8.0.1 ran a Principal Components Analysis (in the attached file "Data Anne for analysis.jasp") that version 0.8.0.2 rejects with the following error message:
[JaspTest.zip](https://github.com/jasp-stats/jasp-desktop/files/1324452/JaspTest.zip)

This analysis terminated unexpectedly.

Error in solve.default(r, result$Structure): system is computationally singular: reciprocal condition number = 4.65224e-18

Stack trace
analysis(dataset = NULL, options = options, perform = perform, callback = the.callback, state = state)

.estimatePCA(dataset, options, perform, nFactor)

psych::principal(dataset, nFactor, rotate = Rotation)

solve(r, result$Structure)

solve.default(r, result$Structure)

**Comment by FransMeerhoff** [comment]
@JohnnyDoorn 

Hi Johnny, can you have a look at this one or redirect it to Alexander?

**Comment by JohnnyDoorn** [comment]
Hi Robert,
I just looked at your files, and tried to reproduce your findings. I did this using JASP version 8.2, which includes the latest bug fixes.
First off, for the Bayesian correlations - which variables did you try to correlate? I did not see any variables labelled A and D. When I put all variables in the Bayesian correlation matrix analysis, The highest Bayes factor equals about 2.6e+13 (between ln_total_pe and ln_ltm_pe). I would recommend always using the latest version of JASP, although I agree that the difference between those Bayes factors is a tad high.  @AlexanderLyNL did anything change in the way the Bayes factors for Pearson's correlation are computed between version 8.0.1 and 8.0.2? 
As for the PCA, I could not reproduce the error - maybe you can try it again using JASP version 8.2 and see if it works now.

**Comment by AlexanderLyNL** [comment]
Yes, we now do a consistency check on the computations. It would be great if we can identify the two variables that led to the discrepancy as this can be used as an additional check. Moreover, I'll do another calculation to check whether the output in our latest release is indeed correct.

**Comment by RobertBiegler** [comment]
Dear Johnny and Alexander

Thanks for the quick response.

First, I noticed that I had reported a problem with version 0.8.0.2 when I meant 0.8.2.0.  Rather important difference.  Sorry.

The attached CSV files show data sets I made up for testing.  The JASP files with the corresponding names and four digits at the end are Bayesian correlations from those data sets, coming from version 0.8.0.0 on my laptop and 0.8.2.0 on my desktop.  Differences show up only for Bayes factors large enough to require exponential notation.  Same goes for the real data set which I put on Github.

I also found that version 0.8.0.0 and 0.8.0.1 are willing to perform a Principal Components Analysis on variables from “ln_ltm_pe” to “cape_positive” inclusive, which version 0.8.2.0 refuses with the error message shown the the attached screen capture.

Regards

Robert Biegler

**Comment by JohnnyDoorn** [comment]
Hi Robert,
Thanks again for taking the time to explain your query. I tried to do the PCA on the variables you indicated, but did not encounter the error - see also the attached screenshot of the output. Does this correspond to your settings? 
![pca_robert](https://user-images.githubusercontent.com/15704203/30972833-f75a9b64-a46b-11e7-877a-c1b4b72987cf.png)

Kind regards
Johnny

**Comment by RobertBiegler** [comment]
Dear Johnny


Ø  I tried to do the PCA on the variables you indicated, but did not encounter the error - see also the attached screenshot of the output. Does this correspond to your settings?

Yes, it does.  I can ask Anne to run the same PCA on her machine, on JASP 0.8.2.0, to see whether the problem is machine specific, or install the newer version on my laptop, and test that way.

Regards

Robert
